R/createParameters.R

```diff
@@ -0,0 +1,461 @@
+  p_specCostRenTang <- p_specCostRen %>%
+    filter(.data[["cost"]] == "tangible") %>%
+    left_join(p_specCostRenTang,
+              by = c("ttot", "reg", "bs", "hs", "bsr", "hsr", "typ", "vin"))
+  p_specCostRenIntang <- p_specCostRen %>%
+    filter(.data[["cost"]] == "intangible") %>%
+    addAssump(brick.file("assump/costIntangRen.csv"))
+  p_specCostRen <- rbind(p_specCostRenTang, p_specCostRenIntang)
+  p_specCostRen <- m$addParameter(
+    "p_specCostRen",
+    c("cost", state, stateR, "vin", "reg", "loc", "typ", "inc", "ttot"),
+    p_specCostRen,
+    description = "floor-space specific renovation cost in USD/m2"
+  )
+
+
+  ## operation ====
+
+  # scenario assumptions
+  carbonPrice <- config[["carbonPrice"]]
+  carrierPriceLevel <- config[["carrierPrices"]]
+  carrierEmiLevel   <- config[["carrierEmi"]]
+
+  ### carbon price ####
+  p_carbonPrice <- if (is.null(carbonPrice)) {
+    data.frame(ttot = ttotNum, value = 0)
+  } else {
+    carbonPrice %>%
+      listToDf() %>%
+      guessColnames(m) %>%
+      toModelResolution(m)
+  }
+  p_carbonPrice <- m$addParameter(
+    "p_carbonPrice",
+    "ttot",
+    p_carbonPrice,
+    description = "Carbon price in USD/t_CO2eq"
+  )
+
+  ### energy carrier prices and emission intensities ####
+  carrierData <- readInput("f_carrierPrices.cs4r",
+                           c("ttot", "reg", "variable", "unit", "carrier", "level"),
+                           inputDir)
+
+  p_carrierPrice <- carrierData %>%
+    filter(.data[["variable"]] == "price") %>%
+    .filterLevel(carrierPriceLevel, "carrierPrice") %>%
+    select("carrier", "reg", "ttot", "value") %>%
+    toModelResolution(m)
+  p_carrierPrice <- m$addParameter(
+    "p_carrierPrice",
+    c("carrier", "reg", "ttot"),
+    p_carrierPrice,
+    description = "final energy carrier price in USD/kWh"
+  )
+
+  p_carrierEmi <- carrierData %>%
+    filter(.data[["variable"]] == "emi") %>%
+    .filterLevel(carrierEmiLevel, "carrierEmi") %>%
+    select("carrier", "reg", "ttot", "value") %>%
+    toModelResolution(m)
+  p_carrierEmi <- m$addParameter(
+    "p_carrierEmi",
+    c("carrier", "reg", "ttot"),
+    p_carrierEmi,
+    description = "energy carrier emission intensity in t_CO2/kWh"
+  )
+
+  ### useful energy demand for space heating ####
+  p_ueDemand <- readInput("f_ueDemand.cs4r",
+                          c("reg", "typ", "vin", "bs", "value"),
+                          inputDir) %>%
+    select("bs", "vin", "reg", "typ", "value") %>%
+    toModelResolution(m)
+  p_ueDemand <- m$addParameter(
+    "p_ueDemand",
+    c("bs", "vin", "reg", "typ"),
+    p_ueDemand,
+    description = "floor-space specific useful energy demand for space heating in kWh/yr/m2"
+  )
+
+  ### FE-to-UE-efficiency of heating systems ####
+  p_eff <- readInput("f_heatingEfficiency.cs4r",
+                     c("ttot", "reg", "hs", "typ", "value"),
+                     inputDir) %>%
+    select("hs", "typ", "reg", "ttot", "value") %>%
```

Comment:
This is a bit inconsistent with other parameters: Usually, ```reg``` comes before ```typ```, see e.g. ```p_ueDemand```

Reply:
Well spotted. Thanks. I changed it.